[Wallet] Changes to work with Xcode 12

cmcewen opened the pull request:
### Description

Updating to Xcode 12 command lines tools requires these changes

Xcode 12 now tries to build both arm64 and x86_64 slices for the simulator
See:
- https://stackoverflow.com/a/63955114/158525
- facebook/react-native#29984
- https://apontious.com/2020/08/23/arm-wrestling-your-ios-simulator-builds/

Also updated fastlane due to an upload issue and CocoaPods to 1.10.0

~~And use Xcode 12.1.0 on CircleCI.~~

### Tested

Successfully ran:

- `yarn test:e2e:ios` (Debug)
- `yarn test:e2e:ios -r` (Release)

Also successful build targeting the Simulator directly within Xcode.

cmcewen committed:
Changes to work with xcode12

jeanregisser commented on packages/mobile/package.json:
```diff
@@ -229,15 +229,15 @@
       "ios.sim.debug": {
         "binaryPath": "ios/build/Build/Products/Debug-iphonesimulator/celo.app",
-        "build": "set -o pipefail && xcodebuild -workspace ios/celo.xcworkspace -scheme celo-test -configuration Debug -sdk iphonesimulator -derivedDataPath ios/build | bundle exec xcpretty",
+        "build": "set -o pipefail && xcodebuild -workspace ios/celo.xcworkspace -scheme celo-test -configuration Debug -derivedDataPath ios/build | bundle exec xcpretty",
```
nit: we can follow up later, but I noticed this change makes it bundle the JS code and assets, so takes more time than it used to.

jeanregisser reviewed (approved):
Thanks 💯

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

cmcewen committed:
podfile

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

jeanregisser committed:
Jeanregisser/xcode12 (#5686)

* Fix Xcode12 cli build

Xcode 12 now tries to build both arm64 and x86_64 slices for the simulator
See:
- https://stackoverflow.com/a/63955114/158525
- https://github.com/facebook/react-native/issues/29984
- https://apontious.com/2020/08/23/arm-wrestling-your-ios-simulator-builds/

Also upgrade CocoaPods

* Use Xcode 12.1.0 on CirclCI

* Fix Android e2e step on CircleCI

See https://support.circleci.com/hc/en-us/articles/360048152911-Java-Builds-Fail-With-Unrecognized-VM-option-UseCGroupMemoryLimitForHeap-

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

jeanregisser committed:
Remove space

jeanregisser committed:
Remove JAVA options

jeanregisser committed:
Try fix android build step

jeanregisser committed:
Fix Android build

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

jeanregisser committed:
Back to Xcode 11 to see if it helps with the iOS e2e test

jeanregisser committed:
Merge remote-tracking branch 'origin/cmcewen/xcode12' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

jeanregisser committed:
Fix iOS e2e build with Xcode 11 (temporary)

jeanregisser committed:
Merge remote-tracking branch 'origin/cmcewen/xcode12' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12

mergify[bot] committed:
Merge branch 'master' into cmcewen/xcode12
